ssas/groups.go

```diff
@@ -257,21 +258,21 @@ type Resource struct {
 	Scopes []string `json:"scopes"`
 }
 
-func GetGroupByGroupID(groupID string) (Group, error) {
+func GetGroupByGroupID(ctx context.Context, groupID string) (Group, error) {
 	var (
 		group Group
 		err   error
 	)
 
-	if err = Connection.First(&group, "group_id = ?", groupID).Error; err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
+	if err = Connection.WithContext(ctx).First(&group, "group_id = ?", groupID).Error; err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
In some places that uses `.first()`, we check for any error that is returned and here we specifically check for the `ErrRecordNotFound` error. Should this be a more generic, catch-all error handling here or do you think this specific error is more appropriate? 🤔

Reply:
We actually return the error below even if it's not an ErrRecordNotFound error, so it is handling all errors. I notice in a couple places we don't actually log or use the error object at all though 😅 Like this API method doesn't use the err:

https://github.com/CMSgov/bcda-ssas-app/blob/9d851236c60174d5e0679b1d3bf1d6792c9c081b/ssas/service/admin/api.go#L680-L682

Reply:
doh, I must have looked right past that. I noticed your second point too! lots of place where that actual err object is not being used..